include/slang/ast/ASTContext.h

```diff
@@ -193,9 +193,12 @@ enum class SLANG_EXPORT ASTFlags : uint64_t {
     DisallowUDNT = 1ull << 44,
 
     /// AST binding is for a bind instantiation (port connection or param value).
-    BindInstantiation = 1ull << 45
+    BindInstantiation = 1ull << 45,
+
+    /// The expression is inside sequence.
+    InsideSequence = 1ull << 46
```

Comment:
You shouldn't need an ASTFlag for this; the checkAssertionCall function is called directly. I would just add another parameter that indicates whether the function return type should be checked or not.

Reply:
fixed